### DISCUSSION TITLE
ftp test connection returns the permission error

### DISCUSSION THREAD

**Discussion by meika** [GetPublii/Publii#1491] [Help]
Just wondering here about the preciseness of the remote pathway  for FTP access :

- /public_html/subdomain/ [ this is what I am using][even if it is wrong why does it not connect and put it there, this user has access to the whole "domain" level structure}
- /home/public_html/subdomain/
- ???? 23.23.23.23/home/public_html/subdomain/

I feel there is some tacit knowledge here someplace...
__________________________________________________________

![image](https://github.com/GetPublii/Publii/assets/245100/e43495ea-77ec-4348-bc36-5e1f8873c602)

_____________

 I can connect via FTP with filezilla just fine, an I am currently uploaded zip files and extracting, but this will get a onerous eventually.
 
This is on nixOS ' which does its thing with Publii but its only Version: 0.42.1 (build 16233).
If the only answer is upgrade the Publii version then I will have to go to NixOS community, but was just wondering what noob error I am making.

There is a discussion some place that I found and lost some weeks ago that the issue may have something to do with IVP6 libraries.

I have seen similar issue in fora but they are on MacOS or otherwise, some issues have gone away by themselves, but this one is continuing. It's been couple of weeks.

(Also plugin for disqus and google search do not function)(the documentation is fine but could do with a cheatsheet or specific checklist across the board once the cookie thing is required, one can loose track of where one is in that process and then one lacks insight into what error or where a misunderstanding occurred). (also I took the prefilled grey serving suggestion words as a prefilled actual field content).

**Comment by internettips** [comment]
Not sure if the general info below will help in your particular instance as you're using a Linux implementation. 

I'm using an Apple Mac. Normally, I tend to use SFTP (with key) rather than the old style FTP.

The 'Remote path' setting is crucial and depends on your web hosting stipulations.

For example, my web host requires the full path for the Remote path setting, which would be:

/home/customer/www/domainname.com/public_html/

in which 'domainname.com' is  the domain name of your website.

If I try to use, say, just /public_html/ or /domainname.com/public_html/ , the SFTP connection will probably fail.

So for my situation, the preciseness of the 'Remote path' is essential.

For IPv6 info, the page at: https://getpublii.com/blog/release-045.html under heading 'Experimental Support for IPv6 with FTP Library' might help.

The Publii 'Recommended Server Settings' information is at: 
https://getpublii.com/docs/recommended-server-settings.html

Maybe other nixOS or Linux users can help further.

**Reply by meika** [reply]
thanks, hopefully will get time in a couple of days (obviously my life is not dependent on it)

**Reply by 3muchfun** [reply]
Hi, @meika ,
tl;dr, to be honest, but from my experience the folder you want Publii to ftp to must exist beforehand.
example: If you put under _Server, FTP with SSL/TLS,_ the Website URL _https://domain.com/subdomain_ , than on the server the folder _**subdomain**_ must exist. At least that solved my problem I had some weeks before.
Not to forget: Under Remote path you need to put in **/subdomain/**

**Comment by meika** [comment]
Okay. I got it to work. 

In my case the  _**precise**_ path is simply the subdomain i.e. /subdomain/

(and yes the folder existed before hand because I was manually doing FTp with a zip file.
